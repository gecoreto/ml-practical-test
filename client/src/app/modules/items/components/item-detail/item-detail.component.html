<div class="item-detail-container p-4">
    <ng-container *ngIf="(working$ | async); else !!(product$ | async) ? content : empty"></ng-container>
    <ng-template #content>
        <div class="item-detail-container-column left">
            <figure class="item-img">
                <img
                    src="https://http2.mlstatic.com/D_NQ_NP_805950-MCO40352919808_012020-O.webp"
                    class="product-item-img__image"
                    alt="Celular iPhone 11 128gb Nuevo 100% Original Y Sellada ">
            </figure>
            <div class="specs">
                <h3 class="specs-title">Descripción del producto</h3>
                <p class="specs-content">
                    CONTENIDO DE LA CAJA:

                -IPHONE
                -CABLE LIGHTNING A USB
                -DOCUMENTACIÓN
        
        
                -Cuerpo:
                Dimensiones: 146.7 x 71.5 x 7.4 mm (5.78 x 2.81 x 0.29 in)
                Peso: 164gr
                Material: Glass front (Gorilla Glass), glass back (Gorilla Glass), aluminum frame
                Apple Pay (Visa, MasterCard, AMEX certified)
                </p>
            </div>
        </div>
        <div class="item-detail-container-column short-description">
            <div class="short-description-title">
                <div class="secodary">
                    <span class="ui-pdp-subtitle">Nuevo  |  10 vendidos</span>
                </div>
                <h3 class="principal">Celular Apple iPhone 12 128gb</h3>
            </div>
            <div class="short-description-price">
                <h1>$ 100.000.00</h1>
            </div>
            <div class="short-description-btns">
                <button
                    class="ml-btn-primary mb-1"
                    type="button">
                    Comprar
                </button>
                <button
                    class="ml-btn-secondary"
                    type="button">
                    Agregar al carrito
                </button>
            </div>
        </div>
    </ng-template>
    <ng-template #empty>
        Producto no encontrado
    </ng-template>
</div>
