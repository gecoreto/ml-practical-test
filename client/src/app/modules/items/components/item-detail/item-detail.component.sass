@import 'utils'

.item-detail-container
  display: flex
  flex-direction: column
  background: var(--ml-white-color)

  &-column
    &.left
      .item-img > img
        width: 100%
        max-width: 500px

    .short-description
      &-btns
        display: flex
        flex-direction: column

  @include media-width-min(md)
    flex-direction: row
    &-column
      &.left
        width: 80%
