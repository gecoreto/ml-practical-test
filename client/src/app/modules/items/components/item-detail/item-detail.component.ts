import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

@Component({
  selector: 'app-item-detail',
  templateUrl: './item-detail.component.html',
  styleUrls: ['./item-detail.component.sass']
})
export class ItemDetailComponent implements OnInit {

  constructor(private route: ActivatedRoute) { }

  ngOnInit(): void {
    // alert(this.route.snapshot.params.id)
  }

}
