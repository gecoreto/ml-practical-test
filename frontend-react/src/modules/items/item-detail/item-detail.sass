@import '../../../sass/utils'

.item-detail-container
  display: flex
  flex-direction: column-reverse
  background: var(--ml-white-color)

  &-column
    &.left
      .item-img > img
        width: 100%
        max-width: 500px

    .short-description
      &-btns
        display: flex
        flex-direction: column
        &.desktop
          display: none

  @include media-width-min(md)
    flex-direction: row
    position: relative
    min-height: 520px
    &-column
      &.left
        width: 80%
      .short-description
        &-btns
          &.mobile
            display: none
          &.desktop
            display: flex

    .card-loader
      &-item-1,
      &-item-2,
      &-item-3
        position: absolute
        left: 10px
      &-item-1
        width: 70%
        height: 300px
      &-item-2
        width: calc( 30% - 30px )
        height: 160px
        left: initial
        right: 10px
      &-item-3
        width: 70%
        top: 340px
