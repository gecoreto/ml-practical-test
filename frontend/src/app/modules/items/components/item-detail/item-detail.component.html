<div class="item-detail-container p-4">
    <ng-container *ngIf="(working$ | async); else !!(product$ | async) ? content : empty">
        <ng-container *ngFor="let i of [1,2,3]">
            <div class="card-loader card-loader-item-{{ i }}"></div>
        </ng-container>
    </ng-container>
    <ng-template #content>
        <ng-container *ngIf="(product$ | async) as item">
            <div class="item-detail-container-column left">
                <figure class="item-img">
                    <img
                        [src]="item.picture"
                        class="product-item-img__image"
                        [alt]="item.title">
                </figure>
                <div class="short-description-btns mobile">
                    <ng-template [ngTemplateOutlet]="buttons"></ng-template>
                </div>
                <div class="specs">
                    <h3 class="specs-title">Descripci√≥n del producto</h3>
                    <p class="specs-content">{{ item.description }}</p>
                </div>
            </div>
            <div class="item-detail-container-column short-description">
                <div class="short-description-title">
                    <div class="secodary">
                        <span class="ui-pdp-subtitle">
                            {{ item.condition }}  |  {{ item.sold_quantity }} vendidos
                        </span>
                    </div>
                    <h3 class="principal">{{ item.title | titlecase }}</h3>
                </div>
                <div class="short-description-price">
                    <h1>
                        {{ item.price.amount | currency:item.price.amount.currency }}
                    </h1>
                </div>
                <div class="short-description-btns desktop">
                    <ng-template [ngTemplateOutlet]="buttons"></ng-template>
                </div>
            </div>
        </ng-container>
    </ng-template>
    <ng-template #empty>Producto no encontrado</ng-template>
    <ng-template #buttons>
        <button
            class="ml-btn-primary mb-1"
            type="button">
            Comprar
        </button>
        <button
            class="ml-btn-secondary"
            type="button">
            Agregar al carrito
        </button>
    </ng-template>
</div>
