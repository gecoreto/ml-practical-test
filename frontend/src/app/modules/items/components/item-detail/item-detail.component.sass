@import 'utils'

.item-detail-container
  display: flex
  flex-direction: column
  background: var(--ml-white-color)

  &-column
    &.left
      .item-img > img
        width: 100%
        max-width: 500px

    .short-description
      &-btns
        display: flex
        flex-direction: column

  @include media-width-min(md)
    flex-direction: row
    position: relative
    min-height: 520px
    &-column
      &.left
        width: 80%

    .card-loader
      &-item-1,
      &-item-2,
      &-item-3
        position: absolute
        left: 10px
      &-item-1
        width: 70%
        height: 300px
      &-item-2
        width: 27%
        height: 160px
        right: 10px
      &-item-3
        width: 70%
        top: 340px
