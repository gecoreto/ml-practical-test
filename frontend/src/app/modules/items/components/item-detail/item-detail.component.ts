import { filter } from 'rxjs/operators';
import { Component, OnDestroy, OnInit } from '@angular/core';
import { ActivatedRoute } from '@angular/router';
import { Observable } from 'rxjs';
import { ItemsI } from '../../entities/items.entity';
import { ItemsFacade } from './../../items.facade';
import { Meta, Title } from '@angular/platform-browser';

@Component({
  selector: 'app-item-detail',
  templateUrl: './item-detail.component.html',
  styleUrls: ['./item-detail.component.sass'],
})
export class ItemDetailComponent implements OnInit, OnDestroy {
  constructor(
    private route: ActivatedRoute,
    private facade: ItemsFacade,
    private meta: Meta,
    private title: Title
  ) {}

  ngOnInit(): void {
    this.facade.fetchProductDetail(this.route.snapshot.params.id);
    this.product$.pipe(filter((item) => !!item)).subscribe((item) => {
      // Para ñadir el título de la página
      this.title.setTitle(item.title);
      // Añadir el tag de la info de la página
      this.meta.addTag({
        name: 'description',
        content: item.description,
      });
    });
  }

  ngOnDestroy(): void {
    // Quitar el tag "description" antes de ir a otra página, para que rastree correctamente la información
    this.meta.removeTag(`name=\'description\'`);
  }

  get product$(): Observable<ItemsI> {
    return this.facade.product$;
  }

  get working$(): Observable<boolean> {
    return this.facade.working$;
  }
}
