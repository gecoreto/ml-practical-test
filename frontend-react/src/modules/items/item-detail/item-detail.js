import React from 'react'
import './item-detail.sass'

export const ItemDetail = () => {
  return(
    <h1>ItemDetail</h1>
  )
}

export default ItemDetail