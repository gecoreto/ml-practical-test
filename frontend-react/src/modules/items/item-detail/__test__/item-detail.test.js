import React from 'react';
import { shallow } from 'enzyme';
import { ItemDetail, Loading } from '../item-detail';

const mockHistoryReplace = jest.fn();
const mockHistoryGo = jest.fn();
const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => {
    return {
        useHistory: () => ({
            push: mockHistoryPush,
            replace: mockHistoryReplace,
            go: mockHistoryGo,
        }),
        useParams: () => ({
            id: '1234'
        }),
    };
});
jest.mock("react-redux", () => {
    return {
        useSelector: () => jest.fn(),
        useDispatch: () => jest.fn()
    };
});

describe("ItemDetail component", () => {

    const wrapper = shallow(<ItemDetail />);

    test('should test ItemDetail component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    describe("WHEN is loading", () => {

        it("THEN Loading should be", () => {
            const wrapper = shallow(<Loading />);
            expect(wrapper).toMatchSnapshot();
        });
    });

});
