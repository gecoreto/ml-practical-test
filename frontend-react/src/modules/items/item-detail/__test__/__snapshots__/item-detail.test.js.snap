// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetail component WHEN is loading THEN Loading should be 1`] = `ShallowWrapper {}`;

exports[`ItemDetail component should test ItemDetail component 1`] = `ShallowWrapper {}`;
